apollo-federation/src/sources/connect/models/validation.rs

```diff
@@ -571,6 +789,22 @@ fn connect_directive_url_coordinate(
     format!("`{http_method}` in `@{connect_directive_name}({CONNECT_HTTP_ARGUMENT_NAME}:)` on `{object}.{field}`")
 }
 
+fn directive_http_header_coordinate(
+    directive_name: &Name,
+    argument_name: &str,
+    object: Option<&Name>,
+    field: Option<&Name>,
+) -> String {
+    match (object, field) {
+        (Some(obj), Some(fld)) => {
```

Comment:
nit:
```suggestion
        (Some(object), Some(field)) => {
```
shadowing variables is common in idiomatic rust